src/GameState/Marble.tsx

```diff
@@ -23,16 +24,21 @@ export const Marble = (props: any) => {
   );
 
   const handleClickMarble = async () => {
-    props.setMarbleClicked({
-      x: props.cellIndex,
-      y: props.rowIndex,
-      value: props.value,
-      isExit: false,
-    });
-    props.setMarbleClickedCoordinates({
-      x: props.cellIndex,
-      y: props.rowIndex,
-    });
+    clicked = !clicked;
```

Comment:
It's a bad bad bad practice to modify the props. And everything will be lost during the next render. In react we use states for this kind of logic.